tests/utils_tests/test_http.py

```diff
@@ -164,6 +164,31 @@ def test_no_allowed_hosts(self):
         # Basic auth without host is not allowed.
         self.assertIs(is_safe_url(r'http://testserver\@example.com'), False)
 
+    def test_is_safe_url_from_settings(self):
+        test_cases = [
+            ('https://testserver/', ['testserver', 'otherserver'], True),
+            ('https://testserver/', ['*'], True),
+            ('https://testserver/', {'*'}, True),
+            ('https://testserver/', ['otherserver', '*'], True),
+            ('https://testserver/', '*', True),
+            ('https://testserver/', [], False),
+            ('https://testserver/', {}, False),
+            ('https://testserver/', ['otherserver'], False),
+            ('https://testserver/', {'otherserver'}, False),
+            ('https://otherserver/', ['testserver', 'otherserver'], True),
+            ('https://otherserver/', ['*'], True),
+            ('https://otherserver/', {'*'}, True),
+            ('https://otherserver/', ['testserver', '*'], True),
+            ('https://otherserver/', '*', True),
+            ('https://otherserver/', [], False),
+            ('https://otherserver/', {}, False),
+            ('https://otherserver/', ['testserver'], False),
+            ('https://otherserver/', {'testserver'}, False),
+        ]
+        for url, allowed_hosts, expected_answer in test_cases:
+            with override_settings(ALLOWED_HOSTS=allowed_hosts):
```

Comment:
Perhaps consider [leveraging subTest](https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests) here?

Reply:
Great suggestion. I didn't know this before.